### ISSUE TITLE
How to run mftf using docker-magento? 

### ISSUE THREAD

**Issue by filmaj** [markshust/docker-magento#189]
Hello dearest docker-magento community,

I've been trying to give running MFTF tests on the instance running inside docker a  shot, to no success thus far. I'm following the [MFTF Getting Started guide](https://devdocs.magento.com/mftf/docs/getting-started.html#run-tests) but I get stumped on a seemingly simple issue: having the test generation bail because it can't reach the magento storefront (over http or https) from inside the container. I get (after exposing some of the URL information a bit  in the error message myself):

```
➜ ./bin/clinotty vendor/bin/mftf generate:tests


In ModuleResolver.php line 444:

  Could not retrieve API token from Magento Instance (Could not find Magento
  Backend Instance at MAGENTO_BACKEND_BASE_URL or MAGENTO_BASE_URL (http://ma
  gento2.test/rest/V1/integration/admin/token))
```

Turns out, I don't know if I can reach the magento storefront from inside the container, regardless of what domain or IP I try to use. 

This is probably because I don't understand how docker-compose works, but figured I'd ask here as someone possibly does know! I think it'd be cool if we could run MFTF tests from inside the container - I think it should be theoretically doable.

**Comment by filmaj** [comment]
Turns out there's multiple containers running different things. The `bin/cli`, `bash`, and `root` commands all by default execute commands in the `phpfpm` container, whereas `nginx`, which is what accepts incoming HTTP requests to the container, runs in the other  `app` container. 

I'm going to continue to explore installing MFTF dependencies, selenium, chrome, etc. in the nginx container and see if I can get MFTF tests running that way.

**Comment by markshust** [comment]
Thanks @filmaj for the report. I'll leave this issue open in case someone else wants to chime in. I can check this out but not sure when I can get around to it.

**Comment by filmaj** [comment]
So I ended up setting up an embedded MFTF setup (as per [MFTF getting started docs](https://devdocs.magento.com/mftf/docs/getting-started.html)) _outside_ of any of this project's containers. That seemed to work for my purposes, though I think that may not work for all purposes as MFTF seems to need access to the magento CLI for certain test scenarios.

Let's keep this issue open as I'll probably come back and continue digging to see if this could be  setup alongside the project.

**Comment by markshust** [comment]
Sure thing, I haven't worked with MFTF except for some dabbling a while back (had issues getting core tests to pass out of the box). Would love to have this integrated into the container.

**Comment by filmaj** [comment]
Agreed, I think it would go a long way to help users as well as I think it could be  a foundation for continuous integration for this project (i.e. every change to this project sets up an instance, configures MFTF, and runs some set of out-of-the-box magento tests to ensure the instance runs as expected).

**Comment by markshust** [comment]
The initial error in this report `Could not find Magento
  Backend Instance at MAGENTO_BACKEND_BASE_URL or MAGENTO_BASE_URL` is most likely caused because Docker cannot find the hostname within the container.

Try again with:
`MAGENTO_BACKEND_BASE_URL=https://host.docker.internal`
?
